### ISSUE TITLE
`TableQuery.schema.createStatements` and `TableQuery.schema.createIfNotExistsStatements` produce different table schemas

### ISSUE THREAD

**Issue by PiotrJander** [slick/slick#2085]
**DESCRIPTION:**

`TableQuery.schema.createStatements` and `TableQuery.schema.createIfNotExistsStatements` produce different table schemas

**STEPS TO REPRODUCE:**

Given the following

```
import slick.jdbc.PostgresProfile.api._
import slick.lifted.ProvenShape

case class Organization(id: Int, name: String, limit: Int)

object Organization {

  // workaround for a Slick bug: https://github.com/VirtusLab/unicorn/issues/11
  val tupled: ((Int, String, Int)) => Organization = (this.apply _).tupled

  class Organizations(tag: Tag)
      extends Table[Organization](
        tag,
        "ORGANIZATIONS"
      ) {

    def id:    Rep[Int]    = column[Int]("ID", O.PrimaryKey, O.AutoInc)
    def name:  Rep[String] = column[String]("NAME", O.Unique)
    def limit: Rep[Int]    = column[Int]("LIMIT")
    override def * : ProvenShape[Organization] =
      (id, name, limit).mapTo[Organization]
  }

  val organizations = TableQuery[Organizations]
}
```

The expression `organizations.schema.createStatements.toList.head` evaluates to

```
create table "ORGANIZATIONS" ("ID" SERIAL NOT NULL PRIMARY KEY,"NAME" VARCHAR NOT NULL UNIQUE,"LIMIT" INTEGER NOT NULL)
```

as expected.

**EXPECTED RESULT:**

I would expect that `organizations.schema.createIfNotExistsStatements.toList.head` should evaluate to

```
create table if not exists "ORGANIZATIONS" ("ID" SERIAL NOT NULL PRIMARY KEY,"NAME" VARCHAR NOT NULL UNIQUE,"LIMIT" INTEGER NOT NULL)
```

**ACTUAL RESULT:**

However, instead it produces

```
create table if not exists "ORGANIZATIONS" ("ID" INTEGER NOT NULL PRIMARY KEY,"NAME" VARCHAR NOT NULL UNIQUE,"LIMIT" INTEGER NOT NULL)
```

Note that the `ID` column gets type `INTEGER` instead of `SERIAL`. This is not what we want with `O.AutoInc` in PostgreSQL.

**SETUP:**

Relevant SBT dependencies include:

```
    "com.typesafe.slick" %% "slick"          % "3.3.1",
    "org.postgresql"     % "postgresql"      % "42.2.5",
```

**Comment by hvesalai** [comment]
Interesting bug, have you had a chance to look at the code that produces this?

**Comment by PiotrJander** [comment]
You mean Slick's source code? Not yet, I'll try to if time allows.

**Comment by PiotrJander** [comment]
I found the culprit: https://github.com/slick/slick/blob/16545a6954960e39c186a9ad146235cee4e4bb48/slick/src/main/scala/slick/jdbc/PostgresProfile.scala#L246 

In my case it could be fixed by removing the `autoIncrement = false` line, but I suppose it is there for a reason, and I'm not sure what else would break then.

In fact, I'm quite disappointed to see that Slick is written in an imperative, mutating style – this sort of bug would not come up in immutable, functional code.

**Comment by hvesalai** [comment]
@PiotrJander hear hear.

I don't understand that line. We need some code archeology to tell us why the method performs differently on the first call compared to following calls.

**Comment by hvesalai** [comment]
The answer is here. The purpose is that when it is set to false, `appendOption` doesn't do some of its magic https://github.com/slick/slick/blob/b0a43942ae2e7a2ebb6fbf048da20fb3e0a1178d/slick/src/main/scala/slick/jdbc/JdbcStatementBuilderComponent.scala#L743

**Comment by hvesalai** [comment]
But the good news is that `appendOption` is protected and it can be modified to take as parameter all the things it depends on. That's the way you can fix this (and then remove the `autoIncrement = false` line (and perhaps make the code immutable).

**Comment by jlncrnt** [comment]
I see the same problem with the following versions:
```
"com.typesafe.slick"   %% "slick"            % "3.3.2"
"org.postgresql"       %  "postgresql"       % "42.2.12"
```
When creating with `createIfNotExists`, even on an empty database, a field like (here with autogen):

`val id: Rep[Int] = column[Int]("id", O.AutoInc, O.PrimaryKey)`

is given a type `INC` and no `nextVal` as default (no autoinc in postgresql).

This result in an error when trying to insert as Slick silently skips the field, and PostgreSQL fails because it does not receive a value.

**Comment by giftig** [comment]
Just to point out since it's not referenced directly here, this is fixed in #2127 I believe, and a duplicate of #2052 which is now closed by that fix.

**Comment by nafg** [comment]
I guess this was fixed in #2130
